Source/qol/autopickup.cpp

```diff
@@ -30,6 +30,23 @@ bool HasRoomForGold()
 	return false;
 }
 
+int numMiscItemsInInv(int iMiscId)
+{
+	int numItems = 0;
+	for (int i = 0; i < Players[MyPlayerId]._pNumInv; i++) {
+		if (Players[MyPlayerId].InvList[i]._iMiscId == iMiscId) {
+			numItems++;
+		}
+	}
+
+	for (int i = 0; i < MAXBELTITEMS; i++) {
+		if (!Players[MyPlayerId].SpdList[i].isEmpty() && Players[MyPlayerId].SpdList[i]._iMiscId == iMiscId) {
+			numItems++;
+		}
+	}
+	return numItems;

```

Comment:
This can use the `InventoryAndBeltPlayerItemsRange` iterator glebm added in #2892
```suggestion
	int numItems = 0;
	for (const auto &item : InventoryAndBeltPlayerItemsRange { Players[MyPlayerId] }) {
		if (item._iMiscId == iMiscId) {
			numItems++;
		}
	}
	return numItems;
```

It handles empty spaces, lets you combine the loops too :)

Reply:
Also just noticed the AutoPickup loop is looking at items on the ground near Players[pnum], while this function is looking only at Players[MyPlayerId]. I think it could desync if a remote player walks near a tile containing a health potion and their client sees they're above the threshold, but the local player doesn't have any health potions in their inventory? Could make this a function on the Player struct so it looks at `this` instead of the global `MyPlayer`